web/src/attribute-filters.tsx

```diff
@@ -288,10 +293,20 @@ export const getMatcherFromSet = (label: string, values: Set<string>): LabelMatc
   }
 
   if (valuesArray.length === 1) {
+    const value = valuesArray[0];
+
+    if (isValueARegex(value)) {
+      return {
+        label,
+        operator: '=~',
```

Comment:
Nit: How about `operator: isValueARegex(value) ? '=~' : '=',` to reduce noise and make it clearer that only the operator depends on whether the value is a regex.